connector/src/main/java/org/jboss/as/connector/subsystems/resourceadapters/ResourceAdapterSubsystemParser.java

```diff
@@ -194,9 +194,10 @@ private void writeRaElement(XMLExtendedStreamWriter streamWriter, ModelNode ra,
         if (transactionSupport == TransactionSupportEnum.XATransaction) {
             isXa = true;
         }
-        if (ra.hasDefined(WM_SECURITY.getName()) && ra.get(WM_SECURITY.getName()).asBoolean()) {
+        if (ra.hasDefined(WM_SECURITY.getName()) && (ra.get(WM_SECURITY.getName()).getType().equals(ModelType.EXPRESSION) || ra.get(WM_SECURITY.getName()).asBoolean())) {
```

Comment:
Maybe let's indeed write it in all circumstances to avoid losing any data that for whatever reason was placed in the config.

Reply:
You mean as having only `(ra.hasDefined(WM_SECURITY.getName())`? Although I'm not against it, that would change the behavior of the subsystem as the configuration is deleted once the `wm-security` attribute is false. I would prefer this issue to deal with the expressions and another issue dealing with the behavior, since this there is downstream dependency for this issue

Reply:
Do we want to change the behavior in this PR or should a next issue for the change be created? @tadamski

Reply:
@petrberan ok, I think creating another issue is fine